### ISSUE TITLE
Sampling does not work in main branch?

### ISSUE THREAD

**Issue by isurulucky** [april-tools/cirkit#331]
Sampling does not seem to work with latest changes in the main branch. Gives the error [1].

```
samples = sampling_query(num_samples=sample_count)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "cirkit/cirkit/backend/torch/queries.py", line 234, in __call__
    samples = self._circuit.evaluate(
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "cirkit/cirkit/backend/torch/graph/modules.py", line 295, in evaluate
    for module, inputs in self._address_book.lookup(module_outputs, in_graph=x):
  File "/cirkit/cirkit/backend/torch/circuits.py", line 58, in lookup
    x = in_graph[..., layer.scope_idx].permute(2, 1, 0, 3)
        ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

EDIT: Not sure if this is only with the latest changes, or if it's an already known issue, as I updated the code to the HEAD of the main after some time.

**Comment by isurulucky** [comment]
Thanks a lot @loreloc !

**Comment by loreloc** [comment]
I should have fixed this in #332 . Let me know if you have other issues.

**Comment by isurulucky** [comment]
> I should have fixed this in #332 . Let me know if you have other issues.

I can confirm that this works. Thanks again @loreloc for the quick fix. 
A related question: are there plans to support conditional sampling, sampling with a marginalized circuit etc., in near future?

**Comment by loreloc** [comment]
Yes, there are plans to support those. Perhaps they will be available in 0.3.0.

**Comment by isurulucky** [comment]
> Yes, there are plans to support those. Perhaps they will be available in 0.3.0.

Great! thank you.
